### ISSUE TITLE
WSL2 does not provide a suitable console binary to open a new console window in

### ISSUE THREAD

**Issue by Swiftloke** [microsoft/WSL#4775]
**Is your feature request related to a problem? Please describe.**

[Pwntools](http://docs.pwntools.com/en/stable/about.html) has useful functionality in which the program can [create a GDB session and attach pwntool's IO functionality to it](http://docs.pwntools.com/en/stable/gdb.html). This GDB session is then opened in a new terminal window and can be interacted with like normal, while your Python script sends and receives data from the process as it runs. However, as far as I can tell, WSL2 provides no way to open a new terminal window [that pwntools defines](http://docs.pwntools.com/en/stable/util/misc.html#pwnlib.util.misc.run_in_new_terminal) (or any way at all). Neither cmd.exe nor bash.exe can perform this task as it launches in the same window as the terminal that spawns it.

**Describe the solution you'd like**

A way for a process to spawn a new terminal window by [``fork``ing and ``execv``ing a new terminal process with the command to be run contained in argv](https://github.com/Gallopsled/pwntools/blob/dev/pwnlib/util/misc.py#L182) or similar.

**Describe alternatives you've considered**

I am uncertain as to whether any solutions already exist or not for this problem. I have been unable to find them, and the WSL issue tracker seems rather open to discussion, so I am posting here in the hopes that I either find a solution and leave a good reference point for any others with this issue, or that a solution is created.

**Additional context**

Running Microsoft Windows [Version 10.0.19041.1]. WSL2 Ubuntu.

``` 
>>> gdb.debug('./bf')
[x] Starting local process '/usr/bin/gdbserver'
[+] Starting local process '/usr/bin/gdbserver': pid 2239
[*] running in new terminal: /usr/bin/gdb -q  "./bf" -x "/tmp/pwn5cwax1eg.gdb"
[ERROR] Could not find a terminal binary to use. Set context.terminal to your terminal.
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/local/lib/python3.6/dist-packages/pwnlib/context/__init__.py", line 1427, in setter
    return function(*a)
  File "/usr/local/lib/python3.6/dist-packages/pwnlib/gdb.py", line 464, in debug
    attach((host, port), exe=exe, gdbscript=gdbscript, need_ptrace_scope = False, ssh=ssh, sysroot=sysroot)
  File "/usr/local/lib/python3.6/dist-packages/pwnlib/context/__init__.py", line 1430, in setter
    return function(*a, **kw)
  File "/usr/local/lib/python3.6/dist-packages/pwnlib/gdb.py", line 779, in attach
    gdb_pid = misc.run_in_new_terminal(cmd)
  File "/usr/local/lib/python3.6/dist-packages/pwnlib/util/misc.py", line 232, in run_in_new_terminal
    log.error('Could not find a terminal binary to use. Set context.terminal to your terminal.')
  File "/usr/local/lib/python3.6/dist-packages/pwnlib/log.py", line 417, in error
    raise PwnlibException(message % args)
pwnlib.exception.PwnlibException: Could not find a terminal binary to use. Set context.terminal to your terminal.```

**Comment by DHowett-MSFT** [comment]
Just throwing thoughts out there: there's no reason `wt.exe` shouldn't be able to fill this need, _once support for command line arguments lands in microsoft/terminal#607_. I can imagine a wrapper that runs `wt.exe -- wsl.exe -d $WSL_DISTRIBUTION -- %s` being the perfect fit here.

Until that works, though, it might need to suffice to use xterm attached to an X server (like X410 or VcXsrv).

**Comment by therealkenc** [comment]
Probably fair enough to call this closed microsoft/terminal#607. The OP is/was a good ask, but not a WSL ask notwithstanding #938.
